mainapp/views.py

```diff
@@ -226,15 +229,33 @@ class Maintenance(TemplateView):
 
 def mapdata(request):
     district = request.GET.get("district", "all")
-    data = cache.get("mapdata:" + district)
-    if data:
-        return JsonResponse(list(data) , safe=False)
-    if district != "all":
-        data = Request.objects.exclude(latlng__exact="").filter(district=district).values()
-    else:
-        data = Request.objects.exclude(latlng__exact="").values()
-    cache.set("mapdata:" + district, data, settings.CACHE_TIMEOUT)
-    return JsonResponse(list(data) , safe=False)
+    page = max(int(request.GET.get("page", "1")), 1)
+    lastid = max(int(request.GET.get("lastid", "0")), 0)
+
+    cachekey = "mapdata:" + district + ',page:' + str(page)+ ',lastid:' + str(lastid)
+    data = cache.get(cachekey)
+    if not data:
+        db = connection.cursor()
+        where, args = '', []
+
+        if district != "all":
+            where += 'where district = %s '
+            args.append(district)
+
+        if lastid:
+            where += 'where id > %s ' if where == '' else 'and id > %s'
```

Comment:
This has to be less than as you are ordering is descending order. So the next request has to server requests less that the given `id`.

Also you will still need the `ORDER BY id DESC LIMIT %s` as you still want to server only a page worth of data.

Reply:
That will only put a person to query another page, fetching data in 1 req vs many, any way lets have both.